resources/js/components/utils/InitializeTanstackQuery.tsx

```diff
@@ -34,25 +37,37 @@ export const InitializeTanstackQuery: ParentComponent = (props) => {
   const t = useLangFunc();
   function toastErrors(error: Error, meta?: Partial<QueryMeta & MutationMeta>) {
     if (!isAxiosError<Api.ErrorResponse>(error)) return;
-    if ((error?.status && error.status >= 500) || !meta?.quietError) {
+    const status = error.response?.status;
+    if (!status || !meta?.quietHTTPStatuses?.includes(status)) {
       let errors = error.response?.data.errors;
       if (meta?.isFormSubmit) {
         // Validation errors will be handled by the form.
         errors = errors?.filter((e) => !Api.isValidationError(e));
       }
       if (errors?.length) {
-        const errorMessages = errors.map((e) =>
-          t(e.code, {
+        const errorMessages = errors.map((e) => {
+          const params = {
             ...(Api.isValidationError(e) ? {attribute: e.field} : undefined),
             ...e.data,
-          }),
-        );
-        for (const msg of errorMessages) {
-          console.warn(`Error toast shown: ${msg}`);
-        }
+          };
+          const translated = () => t(e.code, params);
+          function logWhenAvailable(first = false) {
+            const text = translated();
+            if (text) {
+              console.warn(`Error toast shown: ${text}`);
+            } else {
+              if (first)
+                console.warn(`Error toast shown (translations not ready): ${e.code} ${JSON.stringify(params)}`);
+              // We're not in reactive scope, so use timeout to wait until the translations are available.
+              setTimeout(logWhenAvailable, 500);
```

Comment:
A mógłby toast poczekać na translacje (może z timeoutem) zamiast reaktywności? Jakby userowi miały migać i zmieniać się widoczne toasty to nie wydaje się poprawne UXowo.

Reply:
Coś tam zrobiłem z promisami. Chociaż moim zdaniem reaktywne toasty to nie był akurat istotny problem :) Ale niech będzie, teraz jest chyba w miarę ładnie i w miarę prosto.